website/src/components/Chat/PluginsChooser.tsx

```diff
@@ -0,0 +1,257 @@
+import { useChatContext } from "./ChatContext";
+import { PluginEntry } from "src/types/Chat";
+import { get, post } from "src/lib/api";
+import { API_ROUTES } from "src/lib/routes";
+
+// TODO: Delete me!
+const measureDivWidth = (div: HTMLDivElement | null) => {
+  if (!div) return 0;
+  const width = div?.parentElement?.offsetWidth - 15;
+  return width;
+};
+
+export const PluginsChooser = ({ plugins }: { plugins: PluginEntry[] }) => {
+  const { t } = useTranslation("common");
+  const { control, register, reset, setValue } = useFormContext<ChatConfigFormData>();
+  const [selectedPluginIndex, setSelectedPluginIndex] = useState<number | null>(null);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const textareaRef = useRef<HTMLTextAreaElement | null>(null);
+  const containerRef = useRef<HTMLDivElement | null>(null);
+
+  const findSelectedPluginIndex = useCallback(() => {
+    return plugins.findIndex((plugin) => plugin.enabled);
+  }, [plugins]);
+
+  const handlePluginSelect = useCallback(
+    (index: number) => {
+      const selectedIndex = plugins.findIndex((plugin) => plugin.enabled);
+      const newPlugins = plugins.map((plugin, i) => ({
+        ...plugin,
+        enabled: selectedIndex === index ? false : i === index,
+      }));
+      setValue("plugins", newPlugins, { shouldDirty: true });
+      // onPluginActivated(newPlugins[index].enabled);
+    },
+    [setValue, plugins]
+  );
+
+  const handlePluginEdit = useCallback(
+    (index: number) => {
+      setSelectedPluginIndex(index);
+      onOpen();
+    },
+    [onOpen]
+  );
+
+  const handlePluginRemove = useCallback(
+    (index) => {
+      const updatedPlugins = plugins.filter((_, idx) => idx !== index);
+      setValue("plugins", updatedPlugins, { shouldDirty: true });
+    },
+    [setValue, plugins]
+  );
+
+  const handlePluginSave = useCallback(async () => {
+    if (textareaRef.current) {
+      const newPlugins = [...plugins];
+      newPlugins[selectedPluginIndex!].url = textareaRef.current.value;
+
+      const plugin: PluginEntry = await post(API_ROUTES.GET_PLUGIN_CONFIG, {
+        arg: { plugin: newPlugins[selectedPluginIndex!] },
+      });
+      // disable all existing plugins and enable new one
+      newPlugins.forEach((plugin) => (plugin.enabled = false));
+      newPlugins[selectedPluginIndex!].enabled = true;
+      newPlugins[selectedPluginIndex!].plugin_config = plugin.plugin_config;
+      setValue("plugins", newPlugins, { shouldDirty: true });
+      //onPluginActivated(newPlugins[selectedPluginIndex!].enabled);
+    }
+    onClose();
+  }, [onClose, control, plugins, selectedPluginIndex]);
+
+  const handlePluginAdd = useCallback(() => {
+    const newPlugin: PluginEntry = {
+      url: "",
+      enabled: false,
+    };
+    setValue("plugins", [...plugins, newPlugin]);
+    handlePluginEdit(plugins.length);
+  }, [setValue, plugins, handlePluginEdit]);
+
+  const limitPluginNameLength = (name: string) => {
```

Comment:
seems to me like something we can do with css

Reply:
Yes, it can be done in css, do I create ...module.css file for PluginChooser.tsx or there is another way how CSS is handled in frontend of the OA?

Reply:
oh, I just found out ChakraUI has "isTruncated" and it works fine.
Is that ok to use, from your pov?